tests/unit/plugins/test_support/docker_image_archive_stubbing.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 Red Hat | Ansible
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os.path
+import tarfile
+import tempfile
+
+
+def write_imitation_archive(file_name, image_id, repo_tags):
+    """
+    Write a tar file meeting these requirements:
+
+    * Has a file manifest.json
+    * manifest.json contains a one-element array
+    * The element has a Config property with "[image_id].json" as the value name
+
+    Args:
+        file_name: real file to write
+        image_id: fake sha256 hash (without the sha256: prefix)
+        repo_tags: list of fake image:tag's
+
+    """
+
+    manifest = [
+        {
+            'Config': '%s.json' % image_id,
+            'RepoTags': repo_tags
+        }
+    ]
+
+    write_imitation_archive_with_manifest(file_name, manifest)
+
+
+def write_imitation_archive_with_manifest(file_name, manifest):
+    manifest_file_name = tempfile.mkstemp(suffix='json')[1]
+
```

Comment:
Shouldn't the `try:` start here, since the temp file has already been created and can be removed if the following write fails for whatever reason?

Reply:
Yes, good catch.

I also moved the inner file management inside of the tar file `try..finally`